lib/matplotlib/tests/test_ticker.py

```diff
@@ -159,6 +159,50 @@ def test_SymmetricalLogLocator_set_params():
     nose.tools.assert_equal(sym.numticks, 8)
 
 
+@cleanup
+def test_ScalarFormatter_offset_value():
+    fig, ax = plt.subplots()
+    formatter = ax.get_xaxis().get_major_formatter()
+
+    def check_offset_for(left, right, offset):
+        ax.set_xlim(left, right)
+        # Update ticks.
+        next(ax.get_xaxis().iter_ticks())
+        assert_equal(formatter.offset, offset)
+
+    test_data = [(123, 189, 0),
+                 (-189, -123, 0),
+                 (12341, 12349, 12340),
+                 (-12349, -12341, -12340),
+                 (99999.5, 100010.5, 100000),
+                 (-100010.5, -99999.5, -100000),
+                 (99990.5, 100000.5, 100000),
+                 (-100000.5, -99990.5, -100000),
+                 (1233999, 1234001, 1234000),
+                 (-1234001, -1233999, -1234000),
+                 # Test cases courtesy of @WeatherGod
+                 (.4538, .4578, .45),
+                 (3789.12, 3783.1, 3780),
+                 (45124.3, 45831.75, 45000),
+                 (0.000721, 0.0007243, 0.00072),
+                 (12592.82, 12591.43, 12590),
+                 (9., 12., 0),
+                 (900., 1200., 0),
+                 (1900., 1200., 0),
+                 (0.99, 1.01, 1),
+                 (9.99, 10.01, 10),
+                 (99.99, 100.01, 100),
+                 (5.99, 6.01, 6),
+                 (15.99, 16.01, 16),
+                 (-0.452, 0.492, 0),
+                 (-0.492, 0.492, 0),
+                 (12331.4, 12350.5, 12300),
+                 (-12335.3, 12335.3, 0)]
+
+    for left, right, offset in test_data:
+        yield check_offset_for, left, right, offset
```

Comment:
I _think_ all but two test cases are `left < right`; would it also make sense to yield in the reverse order? Also, I don't think there are any tests for `left == right` (though I don't see that won't work correctly.)